identity.go

```diff
@@ -409,36 +442,59 @@ func (id *ID) queueReload(closeNotify <-chan struct{}) {
 func LoadIdentity(cfg Config) (Identity, error) {
 	id := &ID{
 		Config: cfg,
-		cert:   &tls.Certificate{},
 	}
 
 	var err error
-	id.cert.PrivateKey, err = LoadKey(cfg.Key)
-	if err != nil {
-		return nil, err
+
+	var defaultKey crypto.PrivateKey
+	if cfg.Key != "" {
+		var err error
+		defaultKey, err = LoadKey(cfg.Key)
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	if idCert, err := loadCert(cfg.Cert); err != nil {
-		return id, err
-	} else {
-		id.cert.Certificate = make([][]byte, len(idCert))
-		for i, c := range idCert {
-			id.cert.Certificate[i] = c.Raw
+	// CA bundle is optional, but can be used to fill in the client cert chain
```

Comment:
Interesting. The trust domains are not guaranteed to be the same between the CA bundle and the client certificate.

Client certs from an outside PKI (3rd Party CA) may not have anything in common with the CA bundle. The only time I would expect this to be useful is for client certificates issued by Ziti. And in that case I would hope we set the `cert` field to point to a file w/ the proper chain in it.

This doesn't stop anything from working, but it does allow configurations to work that otherwise wouldn't. It may be confusing that a client through an intermediate CA works because that intermediate CA also happens to be in the ca-bundle.

Reply:
I'm curious to see what kind of interesting edge cases we encounter :)